File: Data-Agent-Server/data-agent-server-app/src/main/java/edu/zsc/ai/domain/service/db/ViewService.java
package edu.zsc.ai.domain.service.db;

import edu.zsc.ai.domain.model.dto.response.db.TableDataResponse;

import java.util.List;

public interface ViewService {

    List<String> listViews(Long connectionId, String catalog, String schema, Long userId);

    String getViewDdl(Long connectionId, String catalog, String schema, String viewName, Long userId);

    void deleteView(Long connectionId, String catalog, String schema, String viewName, Long userId);

    TableDataResponse getViewData(Long connectionId, String catalog, String schema, String viewName, Long userId, Integer currentPage, Integer pageSize);
}

File: Data-Agent-Client/src/services/view.service.ts
import http from '../lib/http';
import { ApiPaths } from '../constants/apiPaths';

export const viewService = {
  listViews: async (connectionId: string, catalog?: string, schema?: string): Promise<string[]> => {
    const params: Record<string, string> = { connectionId };
    if (catalog != null && catalog !== '') params.catalog = catalog;
    if (schema != null && schema !== '') params.schema = schema;

    const response = await http.get<string[]>(ApiPaths.VIEWS, { params });
    return response.data;
  },

  getViewDdl: async (
    connectionId: string,
    viewName: string,
    catalog?: string,
    schema?: string
  ): Promise<string> => {
    const params: Record<string, string> = {
      connectionId,
      viewName,
    };
    if (catalog != null && catalog !== '') params.catalog = catalog;
    if (schema != null && schema !== '') params.schema = schema;

    const response = await http.get<string>(ApiPaths.VIEWS_DDL, { params });
    return response.data;
  },

  deleteView: async (
    connectionId: string,
    viewName: string,
    catalog?: string,
    schema?: string
  ): Promise<void> => {
    await http.delete(ApiPaths.VIEWS, {
      data: {
        connectionId,
        viewName,
        catalog,
        schema
      }
    });
  },
};
